### PR TITLE
Jetpack Content Migration: Tweak Migration Screens UI when fonts are enlarged

### LINKED ISSUE #19919: Jetpack Content Migration: "Please Delete the WordPress app" is covered by the bottom sheet
### Expected behavior

Decrease the space between the description and the image so the latter is visible without the user having to scroll.

### Actual behavior

In small iPhone devices and when dynamic type is enabled, the "Please Delete the WordPress app" image in the last Migration Screen is covered by the bottom action sheet. The user has to scroll to see the image.

### Steps to reproduce the behavior

1. Start the Migration Flow.
2. Head to the Final Migration Screen.
3. Increase Font Sizes ( Enable Dynamic Type ).
4. Notice how the image is covered by the bottom sheet and there is a lot of white space between the image and the description.

### PR DESCRIPTION
**Pull request by salimbraksa**
Fixes #19919

## Description

This PR reduces the gap between the description and the image in the final migration screen.

| Before | After |
| ------ | ----- |
|  |  |

## Test Instructions

**N.B: Since this a UI change, it would be great to test on an iPhone device and iPad.**

#### Notifications Screen
1. Clean install both WordPress and Jetpack
2. Start the Migration Flow
3. Reach the Notifications screen
4. Head to Settings > Accessibility > Display & Text Size
5. Increase font size until the notification fake alert overlaps with the bottom sheet and the view is scrollable
6. **Verify** there is no unnecessary vertical empty space 
7. Decrease font size until the notification fake alert no longer overlaps with the bottom sheet and the view is not scrollable
8. **Verify** the fake alert appears centered between the description and bottom sheet

#### Done Screen
1. From the notifications screen, tap "Continue" to reach the final migration screen
2. Head to Settings > Accessibility > Display & Text Size
3. Increase font size until the "Please delete WordPress" message overlaps with the bottom sheet and the view is scrollable
4. **Verify** there is no unnecessary vertical empty space 
5. Decrease font size until the "Please delete WordPress" message no longer overlaps with the bottom sheet and the view is not scrollable
6. **Verify** the "Please delete WordPress" message appears centered between the description and bottom sheet 

#### Please Delete WordPress Screen
1. Tap "Finish" button
2. While on "My Site" tab, tap "Please Delete WordPress app" to reach "Please Delete WordPress" screen 
3. Head to Settings > Accessibility > Display & Text Size
4. Increase font size until the "Please delete WordPress" message overlaps with the bottom sheet and the view is scrollable
5. **Verify** there is no unnecessary vertical empty space 
6. Decrease font size until the "Please delete WordPress" message no longer overlaps with the bottom sheet and the view is not scrollable
7. **Verify** the "Please delete WordPress" message appears centered between the description and bottom sheet 

## Regression Notes
1. Potential unintended areas of impact

The UI of the following migration screens is impacted, and should be tested:
- The notifications step
- The final step
- "Please delete wordpress" screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationHeaderView.swift [hunk 1]
```diff
@@ -74,10 +74,17 @@ final class MigrationHeaderView: UIView {
...
         imageView.image = configuration.image
         titleLabel.text = configuration.title
         primaryDescriptionLabel.text = configuration.primaryDescription
         secondaryDescriptionLabel.text = configuration.secondaryDescription
+
+        // Hide image and labels if they're empty
+        imageView.isHidden = imageView.image == nil
+        [titleLabel, primaryDescriptionLabel, secondaryDescriptionLabel].forEach { label in
+            label.isHidden = label.text?.isEmpty ?? true
+        }
```
Comment (salimbraksa) [inline WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationHeaderView.swift:87]: Empty labels still take up some space. Hiding them reduce the empty space.

File: WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationStepView.swift [hunk 2]
```diff
@@ -17,17 +17,14 @@ class MigrationStepView: UIView {
     private let centerView: UIView
     private let actionsView: MigrationActionsView
 
-    private lazy var centerContentView: UIView = {
-        let view = UIView()
-        view.addSubview(centerView)
-        return view
-    }()
```
Comment (salimbraksa) [inline WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationStepView.swift:24]: This view was previously added just to have leading and trailing margins for the `centerView`. We could achieve the same without this view. [See `mainStackView` implementation.](https://github.com/wordpress-mobile/WordPress-iOS/pull/19944/files#diff-0a1685b35847103b8ed57f8ae8ebeee09ec41539f7e55fbb37e876c7b0231d8cR24)

File: WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationStepView.swift [hunk 3]
```diff
@@ -17,17 +17,14 @@ class MigrationStepView: UIView {
...
-
     private lazy var mainStackView: UIStackView = {
-        let stackView = UIStackView(arrangedSubviews: [headerView, centerContentView])
+        let stackView = UIStackView(arrangedSubviews: [headerView, centerView, UIView()])
```
Comment (salimbraksa) [inline WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Views/MigrationStepView.swift:21]: The added `UIView()` is to make the `centerView` appear centered in the `UIStackView`.

**Review by twstokes — APPROVED** [review]
Worked great for me @salimbraksa! Great work. 🚀

**Pull request merged**